test/setup-clear-govm.sh

```diff
@@ -7,7 +7,7 @@ set -o pipefail
 
 HOSTNAME=${HOSTNAME:-$1}
 TEST_CRI=${TEST_CRI:-docker}
-TEST_IP_ADDR=${TEST_IP_ADDR:-127.0.0.1}
+TEST_INSECURE_REGISTRIES=${TEST_INSECURE_REGISTRIES:-127.0.0.1:5000}
```

Comment:
This changes the semantic of `TEST_INSECURE_REGISTRIES` from "add additional insecure registries" to "override the default insecure registry".

The Intel-internal pmem-redis BKM was meant to be used with the original semantic: the local, default registry has the PMEM-CSI image, and `TEST_INSECURE_REGISTRIES` is set to enable access to the additional registry which has the pmem-redis image.

I'm leaning towards keeping the original semantic, because then the local registry will always work correctly without extra effort.

@okartau: what do you think?

Reply:
I had not been thinking about needing original and added registry, as I use "override" in my own workflow. But sure, "add additional" is more flexible and thus makes more sense.

Reply:
The default (pulling images from `172.17.0.2`) does not work for me. I'm not sure whether it actually ever did :man_shrugging: 

`172.17.0.2` is the IP address of the first node in the virtual cluster. That's not where I am pushing the images. This comes from `setup-deployment.sh`:
```
TEST_INSECURE_REGISTRIES=${TEST_INSECURE_REGISTRIES:-$(awk '{NF=NF-1;gsub(".*@","");print $NF}' ${WORK_DIRECTORY}/ssh-${CLUSTER}):5000}
...
sed "s|${TEMPLATE_IP}|${TEST_INSECURE_REGISTRIES// */}|g" ${DEPLOYMENT_DIRECTORY}/${deployment_file} | ${KUBECTL} apply -f -
```

Ah, this special interpretation of `TEST_INSECURE_REGISTRIES` was added in 39f05be73416563726d297b6f6a708d28ef703e2. Okay, so one could argue that this is a user error and I should have updated my configuration. @okartau is that how you got it working?

I find this special meaning of the first entry surprising. Nothing in the name suggests that this meaning exists, and it prevents configuring a secure registry to pull the PMEM images from.

My proposal is this:
- Stop using `192.168.8.1:5000` as registry in our deployment files and replace it with `PMEM_REGISTRY`; the fixed IP address used to be a working default when using the `test/qemu-ifup` script, but now it is no longer useful. An explicit variable that is easy to replace in a sed expression is better.
- `ip=$(_work/clear-govm/ssh.0 ip addr show dev ens5 | grep " inet " | sed -e 's/.* inet //' -e 's;/.*;;' -e 's;\.2;.1;')` is the IP address for the host as seen by the cluster. This with port 5000 is always added as insecure registry.
- `TEST_PMEM_REGISTRY` defaults to that same ip:5000, but can be overridden (which seems to be what @okartau needs)
- `TEST_INSECURE_REGISTRIES` adds additional registries (as before)

One more problem with the current code: I ended up with just an IP address in `/etc/docker/daemon.json`, but Docker also wanted the port number there.

@okartau, @chuyd: does this make sense? I can prepare a patch on top of the current branch.

Reply:
seems I created more confusion with my request to combine two registry addresses. Let's go back to two variables if that helps.
About port number: I got cluster created in my workflow by prepending that on cmd line:
`TEST_INSECURE_REGISTRIES=10.123.123.123:5000`
so yes, I used port number there.
But I have noticed current code uses hard-coded 5000 in numerous places, is use of hard-coded port value good idea?

Reply:
My proposal is to hard-code 5000 only in the default value and nowhere else.